facilitator/src/transport/gcs.rs

```diff
@@ -0,0 +1,601 @@
+                        "received HTTP 200 or 201 response with chunks remaining"
+                    ));
+                }
+            }
+            308 => {
+                if !http_response.has("Range") {
+                    return Err(anyhow!(
+                        "No range header in response from GCS: {:?}",
+                        http_response.into_string()
+                    ));
+                }
+                let range_header = http_response.header("Range").unwrap();
+                // The range header is like "bytes=111-222", and represents a
+                // range in the overall object, not the current chunk.
+                let range = range_header
+                    .strip_prefix("bytes=")
+                    .context(format!("unexpected Range header {}", range_header))?;
+                let mut values = range.splitn(2, '-');
+                let start = values
+                    .next()
+                    .context(format!("unexpected Range header {}", range_header))?
+                    .parse::<usize>()
+                    .context(format!("unexpected Range header {}", range_header))?;
+                let end = values
+                    .next()
+                    .context(format!("unexpected Range header {}", range_header))?
+                    .parse::<usize>()
+                    .context(format!("unexpected Range header {}", range_header))?;
+
+                if start != self.object_upload_position {
+                    return Err(anyhow!("unexpected Range header {}", range_header));
+                }
+
+                self.object_upload_position = end + 1;
+
+                // If we have a little content left over, we can't just make
+                // another request, because if there's too little of it, Google
+                // will reject it. Instead, put the portion of the chunk that we
+                // didn't manage to upload back into self.buffer so it can be
+                // handled by a subsequent call to upload_chunk.
+                self.buffer
+                    .extend_from_slice(&chunk[self.object_upload_position - start..]);
+            }
+            _ => {
+                return Err(anyhow!(
+                    "failed to upload part to GCS: {} synthetic: {}\n{:?}",
+                    http_response.status(),
+                    http_response.synthetic(),
+                    http_response.into_string()
+                ))
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl Write for StreamingTransferWriter {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        // Write into memory buffer, and upload to GCS if we have accumulated
+        // enough content
+        self.buffer.extend_from_slice(buf);
+        if self.buffer.len() >= self.minimum_upload_chunk_size {
+            self.upload_chunk(false).map_err(|e| {
+                std::io::Error::new(std::io::ErrorKind::Other, Error::AnyhowError(e))
```

Comment:
Style nit: `use std::io;` at the top of the program, and call these `io::Error::new(..)`